front/src/modules/simulationResult/components/ChartHelpers/drawCurve.ts

```diff
@@ -23,22 +23,41 @@ const drawCurve = <
   const xAxis = getAxis(keyValues, 'x', rotate);
   const yAxis = getAxis(keyValues, 'y', rotate);
 
+  const isHeight = yAxis === 'height';
```

Comment:
I think this bool is more relevant, and then you can use it to change the class and the axis used in the drawZone function
```suggestion
  const onY2Axis = yAxis === 'height' && chart.y2;
```